packages/amplify-e2e-tests/src/__tests__/api_1.test.ts

```diff
@@ -62,6 +63,36 @@ describe('amplify add api (GraphQL)', () => {
     expect(error.message).toContain(`${tableName} not found`);
   });
 
+  it('init a Flutter project and add the simple_model api', async () => {
+    const envName = 'devtest';
+    await initFlutterProjectWithProfile(projRoot, { name: 'simplemodel', envName });
+    await addApiWithSchema(projRoot, 'simple_model.graphql');
+    await amplifyPush(projRoot);
+
+    const meta = getProjectMeta(projRoot);
+    const region = meta.providers.awscloudformation.Region;
+    const { output } = meta.api.simplemodel;
+    const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
+    const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, region);
+
+    expect(GraphQLAPIIdOutput).toBeDefined();
+    expect(GraphQLAPIEndpointOutput).toBeDefined();
+    expect(GraphQLAPIKeyOutput).toBeDefined();
```

Comment:
Check that values are defined before async function call

```ts
    expect(GraphQLAPIIdOutput).toBeDefined();
    expect(GraphQLAPIEndpointOutput).toBeDefined();
    expect(GraphQLAPIKeyOutput).toBeDefined();

    const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, region);
```

Reply:
@jhockett Why do that on line 82 instead of line 76?

Reply:
That way the test can short circuit fail and is easier to debug, because if those `expect` calls fail, it's very clear what went wrong in the test output. If the `getAppSyncApi` call fails because the argument is undefined, it may not be as obvious.

Reply:
@jhockett I pushed your suggestion.